cmd/restic/cmd_rewrite.go

```diff
@@ -152,14 +157,46 @@ func rewriteSnapshot(ctx context.Context, repo *repository.Repository, sn *resti
 				return restic.ID{}, err
 			}
 			ss := querySize()
-			if sn.Summary != nil {
-				sn.Summary.TotalFilesProcessed = ss.FileCount
-				sn.Summary.TotalBytesProcessed = ss.FileSize
+			if sn.Summary == nil { // change of logic: create summary if it wasn't there before
+				sn.Summary = &restic.SnapshotSummary{}
 			}
+			sn.Summary.TotalFilesProcessed = ss.FileCount
+			sn.Summary.TotalBytesProcessed = ss.FileSize
 			return id, err
 		}
 
+	} else if opts.SnapshotSummary {
+		if sn.Summary != nil {
+			Printf("snapshot %s has already got snapshot summary data\n", sn.ID().Str())
```

Comment:
The other options don't print anything for not modified snapshots. Remove the `printf` to be consistent with that behavior.